src/IO/ReadBufferFromMemory.cpp

```diff
@@ -12,7 +12,7 @@ off_t ReadBufferFromMemory::seek(off_t offset, int whence)
 {
     if (whence == SEEK_SET)
     {
-        if (offset >= 0 && internal_buffer.begin() + offset < internal_buffer.end())
```

Comment:
This change seems logical and it allows `seek(0, SEEK_SET)` work for an empty file too.